src/systemctl/systemctl.c

```diff
@@ -2524,6 +2537,10 @@ static int unit_find_paths(
                                         return r;
                         }
                 }
+                else if (!endswith(path, unit_name)) {
```

Comment:
hmm, endswith() doesn't look right here... you probably want `streq(basename(path), unit_name)` here, no?